Assets/Scripts/ThrowController.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ThrowController : MonoBehaviour
+{
+    [Header("Throwing")]
+    [Range(1.0f, 10.0f)] public float sensitivityMouse = 2.0f;
+    [Range(1.0f, 10.0f)] public float sensitivityController = 2.0f;
+    public Transform throwPosition;
+    public float throwableDestroyTime = 5f;
+    public float throwMultiplier = 0.08f;
+    public float minThrowVelocity = 10f;
+    public float maxThrowVelocity = 20f;
+
+    private LaunchArcRenderer launchArcRenderer;
+    private List<GameObject> currentThrowables;
+    private Plane mouseHitPlane;
+    private Camera mainCamera;
+    private bool mainCameraActive = false;
+    private bool isThrowReset = true;
+
+    public delegate void OnThrowEventHandler(Interactable source);
+    public event OnThrowEventHandler OnThrow;
+
+    public delegate void OnPickupEventHandler(GameObject source);
+    public event OnPickupEventHandler OnPickup;
+    
+    public delegate void OnThrowableResetEventHandler();
+    public event OnThrowableResetEventHandler OnThrowableReset;
+
+    public delegate void OnInteractEventHandler(DialogueInteractable source);
+    public event OnInteractEventHandler OnInteractBegin;
+
+    private void Start()
+    {
+        launchArcRenderer = GetComponentInChildren<LaunchArcRenderer>();
+        launchArcRenderer.gameObject.SetActive(false);  // initially hide arc
+        currentThrowables = new List<GameObject>();
+        
+        mouseHitPlane = new Plane(Vector3.up, Vector3.zero);
+        mainCamera = Camera.main;
+
+        UIManager.Instance.staminaBar.OnChange += UpdateThrowVelocity;
+    }
+
+    private void Update()
+    {
+        HandleThrowInput();
+    }
+
+    private void UpdateThrowVelocity(float fillAmount)
+    {
+        launchArcRenderer.velocity = Mathf.Lerp(minThrowVelocity, maxThrowVelocity, fillAmount / StaminaBar.FILL_MAX);
+    }
+
+    public Vector3 GetMousePosition()
+    {
+        if (mainCameraActive)
+        {
+            Vector3 position = transform.position;
+            
+            // Handle controller input
+            if (GameManager.Instance.GetPlayerController().controllerConnected && 
+                !Mathf.Approximately(Input.GetAxis("Horizontal Right Stick") + Input.GetAxis("Vertical Right Stick"),
```

Comment:
Can we add these strings into the Constants file? Especially cause theyre so long

Reply:
I'm leaning towards not putting control names in Constants.cs, but we can talk about it tonight; my reasoning is that the Project Settings --> Input Manager is already doing the same job (assigning fixed strings to actual mouse/controller inputs), so assigning "Horizontal Right Stick" to Constants.HORIZONTAL_RIGHT_STICK wouldn't be helping/making things more consistent.

Reply:
My main reasoning behind putting it in Constants.cs would be in the case we rename anything, we would not have to rename every string occurrence but instead just change the one value in Constants.cs + not having to remember / find exact strings anymore. Maybe we can chat more during the call today about this!